File: longEmbed.js
const { RichEmbed } = require("discord.js");

const longEmbed = function (Client, longTextOrArray, textSeparator, embedTitle, embedColor, pre_or_post) {
    // PRE: textSeparator is appended to the current message, before the split
    // POST: textSeparator is prepended to the following message, after the split
    pre_or_post = typeof pre_or_post !== "undefined" ? pre_or_post : "pre";
    var embed = new RichEmbed();
    if (Client !== null && Client !== undefined) embed.setFooter(Client.user.username, Client.user.avatarURL).setTimestamp();
    if (embedTitle !== null) embed.setTitle(embedTitle);
    if (embedColor !== null) embed.setColor(embedColor);

    if (Array.isArray(longTextOrArray)) {
        var arrayOfText = longTextOrArray;
        var longText = arrayOfText.join(textSeparator);
    } else {
        var reObj = new RegExp(regExpEscape(textSeparator), "g");
        arrayOfText = longTextOrArray.split(reObj);
        longText = longTextOrArray;
    }

    if (longText.length <= 2048) return embed.setDescription(longText);
    else {
        var tmsg = "", msglength = 0, msgcount = 0, m = "", cumulativeMsgLength = 0;
        for (var i = 0; i < arrayOfText.length; i++) {
            if (pre_or_post === "pre") m = arrayOfText[i] + textSeparator;
            else if (pre_or_post === "post") {
                if (i === 0) m = arrayOfText[i];
                else m = textSeparator + arrayOfText[i];
            }
            msglength += m.length;
            if (msgcount > 24 || cumulativeMsgLength + msglength > 6000) {
                // OVER QUOTA
                embed.setFooter("Max number of characters/fields entries reached. There may be more entries, but these were truncated.");
                break;
            }
            if ((msgcount === 0 && msglength < 2048) || (msgcount > 0 && msglength < 1024)) {
                tmsg = tmsg.concat(m);
            } else {
                if (msgcount === 0) embed.setDescription(tmsg);
                else embed.addField("More...", tmsg);
                cumulativeMsgLength += msglength;
                tmsg = m;
                msglength = m.length;
                msgcount++;
            }
        }
        if (!(msgcount > 24 || cumulativeMsgLength + msglength > 6000)) embed.addField("More...", pre_or_post === "pre" ? tmsg.slice(0, -textSeparator.length) : tmsg);
        return embed;
    }
};

module.exports = {
    longEmbed
}

File: longEmbed.py
import discord

def longEmbed(longTextOrArray, textSeparator = "\n", embedTitle = None, embedColor = None):
    if embedTitle is None:
        embedTitle = ""
    if embedColor is None:
        embedColor = 0xe74c3c
    
    if isinstance(longTextOrArray, list):
        arrayOfText = longTextOrArray
        longText = textSeparator.join(longTextOrArray)
    else:
        arrayOfText = longTextOrArray.split(textSeparator)
        longText = longTextOrArray
    
    if len(longText) <= 2048:
        return discord.Embed(title = embedTitle, description = longText, color = embedColor)
    else:
        tmsg = ""
        msglength = 0
        msgcount = 0
        m = ""
        cumulativeMsgLength = 0
        over_quota = False
        for i in range(len(arrayOfText)):
            m = arrayOfText[i] + textSeparator
            msglength += len(m)
            if msgcount > 24 or cumulativeMsgLength + msglength > 6000:
                over_quota = True
                break
            if (msgcount == 0 and msglength < 2048) or (msgcount > 0 and msglength < 1024):
                tmsg += m
            else:
                if msgcount == 0:
                    em = discord.Embed(title = embedTitle, description = tmsg, color = embedColor)
                else:
                    em.add_field(name = "More...", value = tmsg)
                cumulativeMsgLength += msglength
                tmsg = m
                msglength = len(m)
                msgcount += 1
        if not (msgcount > 24 or cumulativeMsgLength + msglength > 6000):
            em.add_field(name = "More...", value = tmsg[:-len(textSeparator)])
        if over_quota:
            em.set_footer(text = "Max number of characters/fields entries reached. There may be more entries, but these were truncated.")
        return em
